### PR TITLE
Show unsupported messages only when failing

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
Hmm, this kinda changes the point of the message. At the moment we kinda set expectations, and with this patch it tries to apologize. When we fail it's not normally because it's a random build, normally it's an actual problem. I think the former is probably better from a support point of view.

**Comment by superm1** [comment]
Something else I was thinking about is maybe a header line for the tool every time you launch it, which I didn't include with this.

Perhaps something like this:
```
# fwupdmgr get-devices
fwupdmgr v1.7.5 (UNSUPPORTED BUILD)
Computer Name
|
|
|--

```

And then supported build removes that string.
Then the warning in the error path makes more sense too.

**Comment by hughsie** [comment]
I'm still not convinced; if we write the firmware successfully and we exit with RC=0 and the device then doesn't work (because of a bug we found before doing the official tarballs) then we never do the apology.

**Comment by superm1** [comment]
> I'm still not convinced; if we write the firmware successfully and we exit with RC=0 and the device then doesn't work (because of a bug we found before doing the official tarballs) then we never do the apology.

I think there is a difference between snap channels though - if someone is using a snap for 1.7.4  in the stable channel vs a snap from 1.7.4 edge channel there are different expectations.  Maybe the answer is somewhere in between?  Stuff that git describes to a tag shouldn't show up front and only apologize on problems and stuff that git describes to not a tag should show up front?

**Comment by hughsie** [comment]
> there are different expectations

Are there? I don't test the snap build before tagging the release, regardless of what channel it's in.

**Comment by hughsie** [comment]
Maybe the right thing to do is make the snap stable channel set the -Dsupported thing?

**Comment by superm1** [comment]
I'm thinking to do it like this, tell me what you think.
1. After we merge meson tristate features branch
2. Make `supported_build` an meson tristate feature
3. By default auto behavior will look at `git describe --exact-match`. If that comes out then automatically set `-Dsupported_build=enabled`.
4. If that fails, then it needs to be overridden by distro packagers (like it does now).

The net result should be no change for anything but snap I think.

**Comment by hughsie** [comment]
I think that makes a lot of sense.

File: contrib/fwupd.spec.in [hunk 1]
```diff
@@ -449,6 +451,7 @@ done
 %{_libdir}/fwupd-plugins-%{fwupdplugin_version}/libfu_plugin_linux_swap.so
 %{_libdir}/fwupd-plugins-%{fwupdplugin_version}/libfu_plugin_linux_tainted.so
 %if 0%{?have_msr}
+%{_libdir}/fwupd-plugins-%{fwupdplugin_version}/libfu_plugin_intel_spi.so
```
Comment (hughsie) [inline contrib/fwupd.spec.in:454]: I deliberately kept this off in the Fedora build as it does unholy things with the PCI bus.

File: contrib/fwupd.spec.in [hunk 2]
```diff
@@ -459,6 +462,7 @@ done
 %{_libdir}/fwupd-plugins-%{fwupdplugin_version}/libfu_plugin_parade_lspcon.so
 %{_libdir}/fwupd-plugins-%{fwupdplugin_version}/libfu_plugin_pci_bcr.so
 %{_libdir}/fwupd-plugins-%{fwupdplugin_version}/libfu_plugin_pci_mei.so
+%{_libdir}/fwupd-plugins-%{fwupdplugin_version}/libfu_plugin_platform_integrity.so
```
Comment (hughsie) [inline contrib/fwupd.spec.in:465]: and this kernel interface never happened, so maybe this wants to be off by default globally

File: contrib/fwupd.spec.in (hunk 1 above)
Comment (superm1) [reply contrib/fwupd.spec.in:454]: so if it's moved to a tristate option - do we perhaps want a new boolean for `experimental_plugins`?

**Comment by superm1** [comment]
With both of those two review points - I am thinking something like create experimental plugins boolean that defaults to false and controls the auto behavior of those two.

File: contrib/fwupd.spec.in (hunk 1 above)
Comment (hughsie) [reply contrib/fwupd.spec.in:454]: I don't think you can clump them together as experimental. e.g. in Fedora I enable the bluez plugin, but have it disabled in RHEL.

**Comment by superm1** [comment]
superseded by https://github.com/fwupd/fwupd/pull/4336

**Pull request closed without merging**